File: make_adder.cpp
#include <iostream>
#include <functional>

std::function<int(int)> makeAdder(int x) {
    return [x](int y) {
        return x + y;
    };
}

int main() {
    auto adderTwo = makeAdder(2);
    std::cout << adderTwo(5) << std::endl;

    int outer = 100;
    {
        int inner = 200;
        std::cout << "Inside block: " << inner << std::endl;
    }

    return 0;
}

File: make_adder.js
function makeAdder(x) {
  return function(y) {
    return x + y;
  };
}

const adderThree = makeAdder(3);
console.log(adderThree(4));
console.log(adderThree("JS"));

function scopeDemo() {
  var a = 10;
  if (true) {
    var a = 20;
    let b = 30;
  }
  console.log(a);
}

scopeDemo();
